added show one carousel and show three carousel

amyshao opened the pull request.

Commit:
added show one carousel and show three carousel

nelson-ens commented on src/components/carousel/show_three.js:
```diff
@@ -0,0 +1,81 @@
+}
+
+window.addEventListener("click", function (event) {
+  var carousel = document.getElementsByClassName("showThreeImages")[0];
```
@amyshao should not rely on first [0], should apply to all, so imagine there are 2 of these on the page...  it needs to run the this script on both...  so u should do a for loop after getting a list of matching divs.

Commit:
added functionality for more than one three-image carousel

nelson-ens reviewed (changes requested):
Please try moving the onClick onto the buttons within the class.

Commit:
redo three image carousel using hide/show instead of inserting, add image border radius

amyshao commented:
Added requested changes and reworked carousel logic to show first 3 images on the DOM instead of inserting left and right images. On 'next' clicked, the first image will be moved to the end. On 'previous' clicked, the last image will be moved to the front.

nelson-ens commented on src/components/carousel/show_three.js:
```diff
@@ -0,0 +1,83 @@
+  }
+}
+
+var allCarousels = [];
```
@amyshao this is no longer needed I believe?

amyshao replied:
Oh cool I guess not since we don't need to loop through all the carousels anymore

nelson-ens commented on src/components/carousel/show_three.js:
```diff
@@ -0,0 +1,83 @@
+function initThreeImageCarousels() {
+  var carousels = document.getElementsByClassName("showThreeImages");
+  for (var i = 0; i < carousels.length; i++) {
+    allCarousels.push(new CarouselWithThreeImages(carousels[i]));
```
@amyshao can probably remove allCarousels.push

nelson-ens commented on src/components/carousel/show_three.js:
```diff
@@ -0,0 +1,83 @@
+var allCarousels = [];
+
+function initThreeImageCarousels() {
+  var carousels = document.getElementsByClassName("showThreeImages");
```
@amyshao we should make this class more specific to target carousels...  perhaps carousel-showthreeimages, so it's more unique to carousels... in case another comp uses the same class...

amyshao replied:
makes sense, I'll name it to carousel-showThreeImages

Commit:
remove carousel array and rename carousel classes

nelson-ensemble commented:
👍

Commit:
merge teaser changes
